src/compression/mod.rs

```diff
@@ -404,8 +404,19 @@ fn mod_p(x: i32, y: i32) -> i32 {
     x - y * div_p(x, y)
 }
 
+use std::cell::Cell;
+thread_local! {
```

Comment:
Have you thought about using `lazy_static`? I wouldn't mind bringing in that extra dependency for code simplification. Could we get rid of `Cell` here, possibly also simplifying the usage of the scratch buffer further down the line?

Also, can we make this more local by introducing

Reply:
`lazy_static!` is global, while we need something thread-local. It also would require a `Mutex` to be mutable, which is more expensive than `Cell`, and no simpler to use.

Reply:
oh right! perfect